misc/dbt-materialize/dbt/include/materialize/macros/relations/columns_spec_ddl.sql

```diff
@@ -17,9 +17,12 @@
 # still tries to parse the constraint, which leads to a confusing error for
 # users.
 {% macro materialize__get_table_columns_and_constraints() -%}
-  {{ exceptions.warn(
-        """
-        Materialize does not support constraints.
-        """
-    )}}
+  {% set assertions = materialize__nullability_assertions() %}
+  {% if not assertions %}
+    {{ exceptions.warn(
+          """
+          Materialize does not support constraints.
```

Comment:
This warning is confusing, since the PR is implementing `NOT NULL` constraints for materialized views.

Reply:
We'll need to rethink this warning at some point, too! See [this comment](https://github.com/MaterializeInc/materialize/pull/21889#discussion_r1355941685).

Reply:
Updated. Open to better wording.